custom_components/localtuya/light.py

```diff
@@ -120,16 +204,28 @@ async def async_turn_on(self, **kwargs):
                 self._lower_brightness,
                 self._upper_brightness,
             )
-            await self._device.set_dp(brightness, self._config.get(CONF_BRIGHTNESS))
+            if self.light_mode == LIGHT_MODES[1]:
+                hs_color = self.hs_color
+                hs_color = (hs_color[0], hs_color[1], int(kwargs[ATTR_BRIGHTNESS]))
+                hs_color = self.hsv_to_tuya_hsv(hs_color)
+                await self._device.set_dp(hs_color, self._config.get(CONF_HS_COLOR))
+            else:
```

Comment:
```suggestion
            await self._device.set_dp(brightness, self._config.get(CONF_BRIGHTNESS))
```
Depending on the light, it might be necessary to set the brightness attribute in addition to the hsv attribute. In any case, it won't do harm to set both.